File: ios-native/NeobankNative/Sources/Core/Config/AppConfig.swift
import Foundation

enum AppConfig {
    static let apiBaseURL: URL = {
        let raw = ProcessInfo.processInfo.environment["API_BASE_URL"] ?? "http://localhost:8080"
        guard let url = URL(string: raw) else {
            fatalError("Invalid API_BASE_URL: \(raw)")
        }
        return url
    }()
}

File: android-native/app/src/main/kotlin/dev/horobets/neobanknative/core/config/AppConfig.kt
package dev.horobets.neobanknative.core.config

import dev.horobets.neobanknative.BuildConfig

/**
 * `API_BASE_URL` is baked in at build time from the environment (see
 * app/build.gradle.kts), mirroring ios-native's `ProcessInfo`-based
 * override. Defaults to the emulator's host-loopback alias, 10.0.2.2,
 * rather than localhost, since the emulator is its own network namespace.
 */
object AppConfig {
    val apiBaseUrl: String = BuildConfig.API_BASE_URL
}
